File: 01/ab.cpp
#include <algorithm>
#include <fstream>

using namespace std;

void calcSum(const int (&array)[3], int &result)
{
    for (size_t i = 0; i < 3; ++i)
    {
        result += array[i];
    }
}

int main()
{
    ifstream input_stream;
    input_stream.open("input.txt");

    string line;
    int sum = 0;

    // Part one
    int maxVal = 0;

    // Part two
    int maxVals[3] = {0, 0, 0};
    int maxValsSum;

    while (getline(input_stream, line, '\n'))
    {
        try
        {
            int val = stoi(line);
            sum += val;
        }
        catch (...)
        {
            // Part one
            maxVal = maxVal > sum ? maxVal : sum;

            // Part two
            maxVals[0] = maxVals[0] > sum ? maxVals[0] : sum;
            sort(maxVals, maxVals + 3);

            sum = 0;
        }
    }

    calcSum(maxVals, maxValsSum);

    printf("\nPart one: %d\n", maxVal);
    printf("Part two: %d\n\n", maxValsSum);

    return 0;
}

File: 01/ab.py
import pandas as pd;

def read_file(file_name):
    try:
        with open(file_name, "r") as file:
            content = file.read()
    except Exception as e:
        print("Something went wrong: ", e)

    return content

def process_input(input):
    data = input.replace('\n', ';').replace(';;', '\n')
    df_transposed = pd.DataFrame([x.split(';') for x in data.splitlines()])
    df = df_transposed.T
    df = df.fillna(0).astype('int')
    return df

def main():
    input = read_file('input.txt')
    data = process_input(input)

    a = data.sum().max()
    b = data.sum().sort_values(ascending=False).head(3).sum()

    print("Part one: {}".format(a))
    print("Part two: {}".format(b))



if __name__ == "__main__":
    main()